### PR TITLE
Enable the typescript compiler.

### PR DESCRIPTION
**Pull request by BramGruneir**
So this enables the typescript transpiler.

Before you call go generate again, install npm and the typescript transpiler:
brew install node
npm install -g typescript

### CODE REVIEW

**Comment by tamird** [comment]
I think you're gonna have to yak shave our docker setup so circleci can run the ts compiler too

File: resource/ui/js/app.js [hunk 1]
```diff
@@ -14,22 +14,23 @@
 // for names of contributors.
 //
 // Author: Andrew Bonventre (andybons@gmail.com)
-
+// Author: Bram Gruneir (bramgruneir@gmail.com)
+/// <reference path="typings/angularjs/angular.d.ts" />
+/// <reference path="typings/angularjs/angular-route.d.ts" />
 var crApp = angular.module('cockroach', [
```
Comment (bdarnell) [inline resource/ui/js/app.js]: We should make it more clear that the .js files are not to be edited. Maybe put these files in ui/js/generated/?

Is it possible to get the typescript compiler to output a comment in the first line starting with "// Generated by "? That's what coffeescript does and github recognizes it to hide the output from github diffs.

File: resource/resource.go [hunk 2]
```diff
@@ -19,6 +19,8 @@
 // and stylesheets.
 package resource
 
+//go:generate tsc -p ui/ts/
```
Comment (bdarnell) [inline resource/resource.go]: Do we want to do this with `go generate`? I think that during development people would rather use go-bindata in debug mode and work on the typescript files without going though `go generate` and `go build`. We should have a command like `make js` to convert from ts to js, and then use `go generate` to convert from js to embedded.go.

**Comment by tschottdorf** [comment]
add to `build/devbase/Dockerfile`:

- apt-get install npm // this can go chained with the "big" apt-get in there
- npm -g install typescript
- ln -s /usr/bin/nodejs /usr/bin/node

File: resource/resource.go (hunk 2 above)
Comment (BramGruneir) [inline resource/resource.go]: For now, I think this makes sense.  We can always convert it to a make js command later on.

File: resource/ui/js/app.js (hunk 1 above)
Comment (BramGruneir) [inline resource/ui/js/app.js]: Done.  Also merged all the files into one giant app.js file.

**Comment by BramGruneir** [comment]
This is fully functional now. PTAL

File: resource/ui/ts/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        "out": "./ui/js/app.js",
+        "sourceMap": false
+    },
+    "files": [
+        "./header.ts",
+        "./app.ts",
+        "./controllers/monitor.ts",
+        "./controllers/rest_explorer.ts"
+    ]
+}
```
Comment (mrtracy) [inline resource/ui/ts/tsconfig.json]: Missing newlines at the end of these files.

Comment (tamird) [inline resource/ui/ts/tsconfig.json]: @BramGruneir there are some configuration options for dealing with whitespace issues in ST. In user settings:
```json
{
	"ensure_newline_at_eof_on_save": true,
        "trim_trailing_white_space_on_save": true
}
```

Comment (BramGruneir) [inline resource/ui/ts/tsconfig.json]: Thanks... and I fixed the files.

**Comment by tamird** [comment]
:+1:

**Pull request merged**